cmux.go

```diff
@@ -177,7 +177,8 @@ func (l muxListener) Accept() (net.Conn, error) {
 // MuxConn wraps a net.Conn and provides transparent sniffing of connection data.
 type MuxConn struct {
 	net.Conn
-	buf buffer
+	buf     bytes.Buffer
```

Comment:
Thanks for the changes! 

I think we can simplify further, by hiding "buf" in "sniffer" (ie bufferReader). If you add the reset() method back, getSniffer() would be something like:

```
if m.sniffer == nil {
    m.sniffer = newBufferReader(m.Conn)
}
m.sniffer.reset()
return m.sniffer
```

WDYT?